app/Models/Asset.php

```diff
@@ -944,6 +961,25 @@ public function getComponentCost(){
      * -----------------------------------------------
      **/
 
+    /**
+     * Make sure the next_audit_date is formatted as Y-m-d.
+     *
+     * This is kind of dumb and confusing, since we already cast it that way AND it's a date field
+     * in the database, but here we are.
+     *
+     * @param $value
+     * @return void
+     */
+    public function getNextAuditDateAttribute($value)
+    {
+        return $this->attributes['next_audit_date'] = $value ? Carbon::parse($value)->format('Y-m-d') : null;
+    }
+
+    public function setNextAuditDateAttribute($value)
+    {
+        $this->attributes['next_audit_date'] = $value ? Carbon::parse($value)->format('Y-m-d') : null;
+    }
+
```

Comment:
I'm hesitant on this change. I think `next_audit_date` might have accidentally changed in #14655 because it's [now](https://github.com/snipe/snipe-it/blob/71b5bf2eefb3669c72b77b61930c1f81b2951dd4/app/Models/Asset.php#L79) `datetime:m-d-Y` which doesn't match the column type.

Can we switch that to be `date` and avoid this addition?

Reply:
I saw this too - and I thought it was weird - but nowhere in our code would `m-d-Y` be a thing. I think that was a mistake, and it's clearly not enforced anyway (lolsob) so I decided not to mess with it for this PR to try to keep the changeset smaller. I agree it should at least be `Y-m-d`, but it sorta seems like Laravel doesn't care much either way. In the docs, the cast only really matters when you're serializing that data, and we already modify it using our data helper (to display the date in whatever way the user has chosen in the admin settings) so I think that's why we never really noticed.